dev/benchmarks/microbenchmarks/lib/foundation/change_notifier_bench.dart

```diff
@@ -6,108 +6,177 @@ import 'package:flutter/foundation.dart';
 
 import '../common.dart';
 
-const int _kNumIterations = 1000;
-const double _scale = 1000.0 / _kNumIterations;
+const int _kNumIterations = 65536;
 const int _kNumWarmUp = 100;
+const int _kScale = 1000;
 
 void main() {
   assert(false, "Don't run benchmarks in checked mode! Use 'flutter run --release'.");
 
-  void listener() {}
-  void listener2() {}
-  void listener3() {}
-  void listener4() {}
-  void listener5() {}
-
-  // Warm up lap
-  for (int i = 0; i < _kNumWarmUp; i += 1) {
-    _Notifier()
-      ..addListener(listener)
-      ..addListener(listener2)
-      ..addListener(listener3)
-      ..addListener(listener4)
-      ..addListener(listener5)
-      ..notify()
-      ..removeListener(listener)
-      ..removeListener(listener2)
-      ..removeListener(listener3)
-      ..removeListener(listener4)
-      ..removeListener(listener5);
-  }
+  // In the following benchmarks, we won't remove the listeners when we don't
+  // want to measure removeListener because we know that everything will be
+  // GC'ed in the end.
```

Comment:
I will add this comment but have you some resources where I could see that it would effectively cause memory leaks?
This is off topic, but in the functional_listener package, listeners are not explicitly removed. I thought that it would cause memory leaks, but I didn't see any when I profiled an app using this package.